weather_mv/loader_pipeline/pipeline.py

```diff
@@ -41,12 +51,73 @@ def pattern_to_uris(match_pattern: str) -> t.Iterable[str]:
         yield from [x.path for x in match.metadata_list]
 
 
+def validate_region(output_table: str, temp_location: t.Optional[str] = None, region: t.Optional[str] = None) -> None:
+    """Validates non-compatible regions scenarios by performing sanity check."""
+
+    def __do_cleanup(bigquery_client: bigquery.Client, storage_client: storage.Client,
+                     canary_output_table: str) -> None:
+        bigquery_client.delete_table(canary_output_table, not_found_ok=True)
+        try:
+            storage_client.get_bucket(CANARY_BUCKET_NAME).delete(force=True)
+        except NotFound:
+            pass
+
+    if not region and not temp_location:
+        raise ValueError('Invalid GCS location: None.')
+
+    bigquery_client = bigquery.Client()
+    storage_client = storage.Client()
+
+    bucket_region = region
+    table_region = None
+
+    if temp_location:
+        parsed_temp_location = urlparse(temp_location)
+        if parsed_temp_location.scheme != 'gs' or parsed_temp_location.netloc == '':
+            raise ValueError(f'Invalid GCS location: {temp_location!r}.')
+        bucket_name = parsed_temp_location.netloc
+        bucket_region = storage_client.get_bucket(bucket_name).location
+
+    try:
+        CANARY_OUTPUT_TABLE = output_table + CANARY_OUTPUT_TABLE_SUFFIX
```

Comment:
`CANARY_OUTPUT_TABLE` is a local and dynamic variable. Thus, it should not be treated as a constant, and should be in snake case.

Reply:
Done.